packages/devkit/src/utils/package-json.ts

```diff
@@ -274,3 +277,68 @@ function requiresRemovingOfPackages(
 
   return needsDepsUpdate || needsDevDepsUpdate;
 }
+
+/**
+ * @typedef EnsurePackageOptions
+ * @type {object}
+ * @property {boolean} dev indicate if the package is a dev dependency
+ * @property {packageManager} string npm, yarn, or pnpm
+ * @property {throwOnMissing} boolean throws an error when the packag is missing
+ * @property {string} cwd the directory to install from
+ * @property {packageJsonPath} string relative path to package.json from workspace root
+ */
+
+/**
+ * Ensure that dependencies and devDependencies from package.json are installed at the required versions.
+ *
+ * For example:
+ * ```typescript
+ * ensureDependencies(tree, {}, { '@nrwl/jest': nxVersion })
+ * ```
+ * This will check that @nrwl/jest@<nxVersion> exists in devDependencies.
+ * If it exists then function returns, otherwise it will install the package before continuing.
+ * When running with --dryRun, the function will throw when dependencies are missing.
+ *
+ * @param tree the file system tree
+ * @param pkg the package to check (e.g. @nrwl/jest)
+ * @param requiredVersion the version to check
+ * @param {EnsurePackageOptions} options
+ * @returns {Promise<void>}
+ */
+export async function ensurePackage(
+  tree: Tree,
+  pkg: string,
+  requiredVersion: string,
+  options: {
+    cwd?: string;
+    dev?: boolean;
+    packageJsonPath?: string;
+    packageManager?: 'npm' | 'yarn' | 'pnpm';
```

Comment:
Should we pass it or should we detect it like we do other places?

Reply:
It's detected, passed for unit tests, but maybe it doesn't matter.